### ISSUE TITLE
False positive for Y_CHECK_PREFER_IS_NOT

### ISSUE THREAD

**Issue by jan-jezek** [SAP/code-pal-for-abap#611]
**Check Name**
Y_CHECK_PREFER_IS_NOT

**Actual Behavior**
Check fires in situations like this:
```abap
IF NOT foo=>bar( ).
 RETURN.
ENDIF.
```
(bar( ) has a returning parameter of type abap_bool)

**Expected Behavior**
The check should not fire.

**Comment by vishalcmahajan-git** [comment]
In your case, the check is firing when you use NOT foo=>bar( ).  To resolve this, consider the following code snippet below:

IF foo=>bar( ) IS NOT INITIAL.
  RETURN.
ENDIF.

**Comment by jelliottp** [comment]
> In your case, the check is firing when you use NOT foo=>bar( ). To resolve this, consider the following code snippet below:
> 
> IF foo=>bar( ) IS NOT INITIAL. RETURN. ENDIF.

In my opinion this would not be preferred. This is for methods that return an `abap_bool` and `IF NOT foo=>bar( ).` is the shorthand [predicative method call](https://github.com/SAP/styleguides/blob/main/clean-abap/CleanABAP.md#consider-using-predicative-method-calls-for-boolean-methods). 

Also, these methods are(or should be) typically named like `is_condition_met()` or `has_some_data()`, so it doesn't make sense with natural language to use `IF is_condition_met() IS NOT INITIAL`.

**Comment by larsip** [comment]
Hallo,

I don't think that this is still a issue?

I have implemented this small report and I don't get the false positive. 
I remember I also have it earlier, but I think it have be fixed with #599.

```abap
REPORT ztest_prefer_is_not.

CLASS foo DEFINITION FINAL.
  PUBLIC SECTION.
    CLASS-METHODS bar RETURNING VALUE(rv_result) TYPE abap_bool.

    METHODS test.
ENDCLASS.


CLASS foo IMPLEMENTATION.
  METHOD bar.
    rv_result = abap_true.
  ENDMETHOD.

  METHOD test.
    IF NOT foo=>bar( ).
      RETURN.
    ENDIF.
  ENDMETHOD.
ENDCLASS.
```

**Comment by bjoern-jueliger-sap** [comment]
Seems to be right, this should have been fixed with #599 and I can't reproduce it either - however this issue is younger than #599. Perhaps OP used an old version.
